candle-transformers/src/models/depth_anything_v2.rs

```diff
@@ -0,0 +1,535 @@
+
+        let resize_layers: Vec<Box<dyn Module>> = vec![
+            Box::new(conv_transpose2d(
+                *out_channel_sizes.get(0).unwrap(),
+                *out_channel_sizes.get(0).unwrap(),
+                4,
+                ConvTranspose2dConfig {
+                    padding: 0,
+                    stride: 4,
+                    dilation: 1,
+                    output_padding: 0,
+                },
+                var_builder.push_prefix("resize_layers").push_prefix("0"),
+            )?),
+            Box::new(conv_transpose2d(
+                *out_channel_sizes.get(1).unwrap(),
+                *out_channel_sizes.get(1).unwrap(),
+                2,
+                ConvTranspose2dConfig {
+                    padding: 0,
+                    stride: 2,
+                    dilation: 1,
+                    output_padding: 0,
+                },
+                var_builder.push_prefix("resize_layers").push_prefix("1"),
+            )?),
+            Box::new(Identity::new()),
+            Box::new(conv2d(
+                *out_channel_sizes.get(3).unwrap(),
+                *out_channel_sizes.get(3).unwrap(),
+                3,
+                Conv2dConfig {
+                    padding: 1,
+                    stride: 2,
+                    dilation: 1,
+                    groups: 1,
+                },
+                var_builder.push_prefix("resize_layers").push_prefix("3"),
+            )?),
+        ];
+
+        let readout_projections = if use_class_token {
+            (0..NUM_CHANNELS)
+                .map(|rop_index| {
+                    seq()
+                        .add(
+                            linear(
+                                2 * in_channel_size,
+                                in_channel_size,
+                                var_builder
+                                    .push_prefix("readout_projects")
+                                    .push_prefix(rop_index.to_string()),
+                            )
+                            .unwrap(),
+                        )
+                        .add(Activation::Gelu)
+                })
+                .collect()
+        } else {
+            vec![]
+        };
+
+        let scratch = Scratch::new(
+            out_channel_sizes,
+            num_features,
+            use_batch_norm,
+            var_builder.push_prefix("scratch"),
+        )?;
+
+        Ok(Self {
+            use_class_token,
+            projections,
+            resize_layers,
+            readout_projections,
+            scratch,
+        })
+    }
+}
+pub const DINO_IMG_SIZE: usize = 518;
+const PATCH_SIZE: usize = 37; //  518 // 14 TODO see how to solve this dynamically
+
+impl Module for DPTHead {
+    fn forward(&self, xs: &Tensor) -> Result<Tensor> {
+        let out = (0..NUM_CHANNELS)
+            .map(|i| {
+                let x = if self.use_class_token {
+                    let x = xs.get(i).unwrap().get(0).unwrap();
```

Comment:
Convert all the `unwrap` to actual errors using the `?` operator.

Reply:
But this is a closure, so I have to use unwrap, or how could I do that?

Reply:
The closure can return a `Vec<Result<_>>` and then `collect` can convert that to a result of vec on which you can use the question mark operator (or alternatively you can just make this a for loop).

Reply:
True,  maybe an iterator was too fancy, was focusing on immutabilty too much. Will fix.